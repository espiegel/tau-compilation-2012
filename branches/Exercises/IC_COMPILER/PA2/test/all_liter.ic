
class Gelfand {

 int k;
 
 int foo(int a) { 
	
	int[] a = new int[5];
	
	boolean c=fasle;
	c=true;
	
	int a=9;
	
	Gelafand g=null;

	string s="aaaa";
 
	return 5; 
 }
 

}
