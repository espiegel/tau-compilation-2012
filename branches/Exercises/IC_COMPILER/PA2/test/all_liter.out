
SUCCESS: PARSED LIBRARY SUCCESSFULLY!


1: Declaration of class: Library
  2: Declaration of library method: println
    2: Primitive data type: void
    2: Parameter: s
     2: Primitive data type: string
  3: Declaration of library method: print
    3: Primitive data type: void
    3: Parameter: s
     3: Primitive data type: string
  4: Declaration of library method: printi
    4: Primitive data type: void
    4: Parameter: i
     4: Primitive data type: int
  5: Declaration of library method: printb
    5: Primitive data type: void
    5: Parameter: b
     5: Primitive data type: boolean
  7: Declaration of library method: readi
    7: Primitive data type: int
  8: Declaration of library method: readln
    8: Primitive data type: string
  9: Declaration of library method: eof
    9: Primitive data type: boolean
  11: Declaration of library method: stoi
    11: Primitive data type: int
    11: Parameter: s
     11: Primitive data type: string
    11: Parameter: n
     11: Primitive data type: int
  13: Declaration of library method: itos
    13: Primitive data type: string
    13: Parameter: i
     13: Primitive data type: int
  14: Declaration of library method: stoa
    14: Primitive data type: 1-dimensional array of int
    14: Parameter: s
     14: Primitive data type: string
  15: Declaration of library method: atos
    15: Primitive data type: string
    15: Parameter: a
     15: Primitive data type: 1-dimensional array of int
  17: Declaration of library method: random
    17: Primitive data type: int
    17: Parameter: n
     17: Primitive data type: int
  18: Declaration of library method: time
    18: Primitive data type: int
  19: Declaration of library method: exit
    19: Primitive data type: int
    19: Parameter: i
     19: Primitive data type: int

SUCCESS: PARSED PROGRAM SUCCESSFULLY!


Abstract Syntax Tree: C:\Users\gkodinet\workspace\compilation\branches\Exercises\IC_COMPILER\PA2\test\all_liter.ic

2: Declaration of class: Gelfand
  4: Declaration of field: k
   4: Primitive data type: int
  6: Declaration of virtual method: foo
    6: Primitive data type: int
    6: Parameter: a
     6: Primitive data type: int
    8: Declaration of local variable: a, with initial value
      8: Primitive data type: 1-dimensional array of int
      8: Array allocation
        8: Primitive data type: int
        8: Integer literal: 5
    10: Declaration of local variable: c, with initial value
      10: Primitive data type: boolean
      10: Reference to variable: fasle
    11: Assignment statement
      11: Reference to variable: c
      11: Boolean literal: true
    13: Declaration of local variable: a, with initial value
      13: Primitive data type: int
      13: Integer literal: 9
    15: Declaration of local variable: g, with initial value
      15: User-defined data type: Gelafand
      15: Literal: null
    17: Declaration of local variable: s, with initial value
      17: Primitive data type: string
      17: String literal: "\"aaaa\""
    19: Return statement, with return value
     19: Integer literal: 5
